### ISSUE TITLE
Test failure: ResponseCompression_AppAndHostCompression

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/ServerTests#146]
This test [fails](http://aspnetci/viewLog.html?buildId=557737&buildTypeId=XPlat_Windows_Win8_Universe) occasionally with the following error:
```
System.Net.Http.HttpRequestException : An error occurred while sending the request.
---- System.IO.IOException : Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.
-------- System.Net.Sockets.SocketException : An existing connection was forcibly closed by the remote host
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IntegrationTesting.LoggingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHelper.RetryRequest(Func`1 retryBlock, ILogger logger, CancellationToken cancellationToken, Int32 retryCount)
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHelper.RetryRequest(Func`1 retryBlock, ILogger logger, CancellationToken cancellationToken, Int32 retryCount)
   at ServerComparison.FunctionalTests.ResponseCompressionTests.ResponseCompression(TestVariant variant, Func`3 scenario, Boolean hostCompression, String testName) in /_/test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs:line 142
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.FillAsync()
   at System.Net.Http.HttpConnection.ReadNextResponseHeaderLineAsync(Boolean foldedHeadersAllowed)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
----- Inner Stack Trace -----
------- Stdout: -------
| [0.001s] TestLifetime Information: Starting test ResponseCompression_AppAndHostCompression-Server: Kestrel, TFM: net461, Type: Standalone, Arch: x64 at 2018-09-20T18:46:14
| [0.001s] TestLifetime Information: Starting test ResponseCompression_AppAndHostCompression_Kestrel_net461_x64_Standalone at 2018-09-20T18:46:14
| [0.001s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: Deploying [Variation] :: ServerType=Kestrel, Runtime=Clr, Arch=x64, BaseUrlHint=, Publish=False
| [0.001s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: Executing C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe  --urls http://127.0.0.1:0/ --server Microsoft.AspNetCore.Server.Kestrel
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: SET ASPNETCORE_ENVIRONMENT=ResponseCompression
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: SET ASPNETCORE_DETAILEDERRORS=true
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: SET ASPNETCORE_CONTENTROOT=C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites
| [0.003s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: Started C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe. Process Id : 7448
| [0.374s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: Hosting environment: ResponseCompression
| [0.374s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: Content root path: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites
| [0.374s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: Now listening on: http://127.0.0.1:54034
| [0.375s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: Application started. Press Ctrl+C to shut down.
| [0.375s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: Application ready at URL: http://127.0.0.1:54034/
| [0.375s] ResponseCompression Warning: Retry count 1..
| [0.503s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: info: Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets[14]
| [0.503s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout:       Connection id "0HLGV55BUEB52" communication error.
| [0.503s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @Eilon (because the bot doesn't know who else to pick)

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
ResponseCompression_AppAndHostCompression [failed with about the same error](http://aspnetci/viewLog.html?buildId=557737&buildTypeId=XPlat_Windows_Win8_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
ResponseCompression_AppAndHostCompression [failed with about the same error](http://aspnetci/viewLog.html?buildId=558038&buildTypeId=XPlat_Windows_Win8_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by Tratcher** [comment]
Same issue as elsewhere:
```
System.IO.FileLoadException: Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
| [0.666s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: File name: 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

https://github.com/aspnet/Home/issues/3544

**Comment by aspnet-hello** [comment]
There were 7 failures [with about the same error](http://aspnetci/viewLog.html?buildId=558745&buildTypeId=Lite_UniverseTest) on master:
- HelloWorld
- ResponseCompression_AppAndHostCompression
- HelloWorld
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppAndHostCompression


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 7 failures [with about the same error](http://aspnetci/viewLog.html?buildId=559368&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- ResponseCompression_AppCompression
- HelloWorld
- HelloWorld
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppAndHostCompression


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by muratg** [comment]
Looks like this is ongoing. @Eilon @mkArtakMSFT is there someone who can look into Wes' recommendation here: https://github.com/aspnet/Home/issues/3544#issuecomment-423692971?

**Comment by aspnet-hello** [comment]
There were 7 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560048&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- HelloWorld
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppCompression
- HelloWorld


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 7 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560221&buildTypeId=XPlat_Windows_Win7_Universe) on master:
- ResponseCompression_AppCompression
- HelloWorld
- ResponseCompression_AppAndHostCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppAndHostCompression
- HelloWorld


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 7 failures [with about the same error](http://aspnetci/viewLog.html?buildId=561382&buildTypeId=Lite_UniverseTest) on master:
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_AppAndHostCompression
- ResponseCompression_NoCompression
- HelloWorld
- HelloWorld
- ResponseCompression_AppCompression


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
This was solved when @natemcmaster fixes aspnet/Home#3544.
